address.md

```diff
@@ -0,0 +1,262 @@
+
+#### Design criteria
+
+1. **Identifiable**: The address must be easily identifiable as a Filecoin address.
+2. **Reliable**: Addresses must provide a mechanism for error detection when they might be transmitted outside the network.
+3. **Upgradable**: Addresses must be versioned to permit the introduction of new address formats.
+4. **Compact**: Given the above constraints, addresses must be as short as possible.
+
+## Specification
+
+A Filecoin address is a byte string consisting of:
+* A **network indicator** that identifies which network this address belongs to.
+* A **protocol indicator** that identifies the type and version of this address.
+* The **payload** used to uniquely identify the actor according to the protocol.
+* A **checksum** verifies the address is valid.
+
+```
+|---------|----------|---------|----------|
+| network | protocol | payload | checksum |
+|---------|----------|---------|----------|
+| 1 byte  |  1 byte  | n bytes |  4 bytes |
+```
+
+An example of a Filecoin address in golang:
+
+```go
+type Address struct {
+    // 0: main-net
+    // 1: test-net
+    network byte
+    
+    // 0: ID
+    // 1: Blake2b160-Hash of secp256k1 Public Key
+    // 2: Blake2b160-Hash of Actor creation data
+    // 3: BLS Public Key
+    protocol byte
+    
+    // raw bytes containing the data associated with protocol
+    payload []byte
+}
+```
+
+#### Network Indicator
+
+The **network** byte identifies the network the address belongs to:
+
+- `0` : Mainnet
+- `1` : Testnet
+
+An example description in golang:
+
+```go
+// String Prefix
+Mainnet_Prefix = 'f'
+Testnet_Prefix = 't'
+
+type Protocol = byte
+const (
+    Mainnet Network = iota
+    Testnet
+)
+```
+
+#### Protocol Indicator
+
+The **protocol indicator** byte describes how a method should interpret the information in the payload field of an address. Any deviation for the algorithms and data types specified by the protocol must be assigned a new protocol number. In this way, protocols also act as versions.
+
+- `0`: ID
+- `1` : SECP256K1 Public Key
+- `2` : Actor
+- `3` : BLS Public Key
+
+An example description in golang:
+
+```go
+// Protocol byte
+type Protocol = byte
+const (
+    ID Protocol = iota
+    SECP256K1
+    Actor
+    BLS
+)
+```
+
+###### Protocol 0: IDs
+
+**Protocol 0** addresses are simple ids. These addresses are expected to remain within the network. All actors have a numeric ID even if they don't have public keys. The key part of an ID address is the decimal representation of the id only characters '0'-'9'. IDs are not hashed and do not have a checksum.
```

Comment:
This sentence is confusing to me:
```
The key part of an ID address is the decimal representation of the id only characters '0'-'9'.
```

Reply:
Yeah, worth just saying its encoded base 10